src/kedro_vizro_labeling/dashboards/components.py

```diff
@@ -32,7 +33,9 @@ class Modal(vm.VizroBaseModel):
             underlying component may change in the future. Defaults to `{}`.
 
     """
-
+    # AM comment: do we really want Containers here for header, body, footer? It's very flexible since you can put
+    # any component inside but feels potentially like overkill compared to just allowing markdown text and probably
+    # makes the formatting of the modal a little off in terms of spacing.
```

Comment:
I wanted the footer to contain buttons to log in and I felt maybe I would have wanted to show an icon in the header (I guess you can do that with vm.Text?).

I also created this Modal with other use cases in mind. In the case of labeling, when an event is selected, the user will have to click on a button to open a model which has a body that will contain a form to fill.

But indeed, I did not think at all of the potential spacing issues.

Reply:
Makes sense 👍 You can't do an icon with `vm.Text` but if it's a common requirement then you might like to have a whole `icon` field dedicated to it.

If this became a built-in Vizro component then we'd need to re-think exactly what functionality we want to expose, what fields to have and how to configure the model. But if it's just a custom component that you use yourself then of course you should use whatever works for you! So if you need the full flexibility of `vm.Container` then it's a good idea, and you can probably fix the excessive spacing with some CSS without too much difficulty.

Reply:
Tweaking CSS properties is always a pain for me unfortunately, so I'll keep that in mind.